base_layer/core/src/mempool/priority/prioritized_transaction.rs

```diff
@@ -38,12 +41,12 @@ use crate::transactions::{transaction_components::Transaction, weight::Transacti
 pub struct FeePriority(Vec<u8>);
 
 impl FeePriority {
-    pub fn new(transaction: &Transaction, insert_epoch: u64, weight: u64) -> Self {
+    pub fn new(transaction: &Transaction, insert_epoch: u64, weight: u64) -> Result<Self, TransactionError> {
         // The weights have been normalised, so the fee priority is now equal to the fee per gram ± a few pct points
         // Include 3 decimal places before flooring
         #[allow(clippy::cast_possible_truncation)]
         #[allow(clippy::cast_sign_loss)]
-        let fee_per_byte = ((transaction.body.get_total_fee().as_u64() as f64 / weight as f64) * 1000.0) as u64;
+        let fee_per_byte = ((transaction.body.get_total_fee()?.as_u64() as f64 / weight as f64) * 1000.0) as u64;
```

Comment:
since we are multiplying by 10000 anyway, why can't we remove the conversion to f64?

Reply:
also, can't the *1000 cause an overflow here?

Reply:
overflow is not possible as total fee is a u64 converted to f64
u64::max = 1.8 E^19
f64::max = 1.79 E^308

So there are 2 options are, keep as is with f64
Or do
```
((transaction.body.get_total_fee().as_u64() as u128 *1000) /weight) as u64
```